Multiplayer networking renames/simplification

mhilbrunner opened the pull request:
Removes the **networking** prefix from some methods and members, now that multiplayer has been largely moved out of Node and SceneTree and is seperated into its own set of classes. Previously, it was useful to mark these so people knew they were multiplayer related when looking at them on **Node** or **SceneTree**, but now they have moved to **Multiplayer** its self-evident and just makes them longer and less readable.

Part of https://github.com/godotengine/godot/issues/16863.

----

**RENAMES**

**MultiplayerAPI**
**Functions and Properties**
`multiplayer.is_network_server()` -> `multiplayer.is_server()`

`multiplayer.network_peer` -> `multiplayer.multiplayer_peer`
`multiplayer.has_network_peer()` ->`multiplayer.has_multiplayer_peer()`
`multiplayer.get_network_peer()` -> `multiplayer.get_multiplayer_peer()`
`multiplayer.set_network_peer()` -> `multiplayer.set_multiplayer_peer()`

`multiplayer.get_network_connected_peers()` -> `multiplayer.get_peers()`

`multiplayer.get_network_unique_id` -> `multiplayer.get_unique_id()`

`multiplayer.refuse_new_network_connections` -> `multiplayer.refuse_new_connections`
`multiplayer.set_refuse_new_network_connections()` -> `multiplayer.set_refuse_new_connections()`
`multiplayer.is_refusing_new_network_connections()` -> `multiplayer.is_refusing_new_connections`

**Signals**
`network_peer_connected` -> `peer_connected`
`network_peer_disconnected` -> `peer_disconnected`
`network_peer_packet` -> `peer_packet`

**Node**
**Functions and Properties**

`Node.network_authority` -> `Node.multiplayer_authority`
`Node.is_network_authority()` -> `Node.is_multiplayer_authority()`
`Node.set_network_authority()` -> `Node.set_multiplayer_authority()`
`Node.get_network_authority()` -> `Node.get_multiplayer_authority()`

----

Note this contains very minor changes to some log messages that are now cleaned up in `RPCManager::_send_rpc()`. I had to edit a few of them anyway for consistency with these changes.

mhilbrunner committed:
Multiplayer networking renames/simplification

Removes _networking_ prefix from some methods and members, now that multiplayer has been largely moved out of Node and SceneTree and is seperated into its own set of classes.

alexdevteam commented:
+1 for get_peers, connection is implied as you mentioned.

mhilbrunner commented:
Note this contains very minor changes to some log messages that are now cleaned up in `RPCManager::_send_rpc()`. I had to edit a few of them anyway for consistency with these changes.

Faless commented:
Reaming of `get/has_network_peer` to `has/get_peer` could be a bit confusing (i.e. with `has_peer(peer_id)`),
And I think `get_connected_peers` is like that for the same reason (i.e. distinguishing between what we call a `network_peer` -> `MultiplayerPeer` instance, and a `connected_peer` or simply `peer`, meaning a remote peer ID).

So, I would leave `get/has_network_peer` untouched, but I'm okay with renaming `get_connected_peers` to just `get_peers`.
Alternatively, we could rename `get/has_network_peer` to `get/has_multiplayer_peer` (staying consistent with the idea that the result is a `MultiplayerPeer` instance or null),

mhilbrunner commented:
We discussed this briefly and decided to indeed rename `multiplayer.get_network_connected_peers()` to just `multiplayer.get_peers()`.

`get/has_network_peer` will become `get/has_multiplayer_peer`, and we'll change `network authority` to `multiplayer authority` for consistency. Will do so tomorrow :)

mhilbrunner commented:
@Faless Updated as discussed, `MultiplayerAPI.(set/has/get_)network_peer` is now `MultiplayerAPI.(set/has/get_)multiplayer_peer`, `Node.(set/get_)network_authority` is now `Node.(set/get_)multiplayer_authority` and `MultiplayerAPI.get_network_connected_peers()` is now just `MultiplayerAPI.get_peers()`.

Also updated the PR description to reflect these changes.

Faless reviewed (approved):
Looking good! :+1: . Thanks!

nathanfranke commented:
```py
var net := ENetMultiplayerPeer.new()
var err := net.create_server(12345)
assert(err == OK)
get_tree().multiplayer.multiplayer_peer = net
```

`get_tree().multiplayer.multiplayer_peer = net` is pretty verbose, what about `get_tree().multiplayer.peer = net`? Would that be confusing compared to `get_peers()`? Or am I just nitpicking?

Faless commented:
@nathanfranke you don't need to call `get_tree()` if you are not using the custom multiplayer override. You can just call `multiplayer`.
`multiplayer.set_multiplayer_peer(peer)` or `multiplayer.multiplayer_peer = net`.

mhilbrunner commented:
Yeah, I had changed it to just peer first as you suggest, but it was indeed unclear because of the peer functions on multiplayer that aren't related, thus we changed it to multiplayer_peer. Its verbose, but usually you shouldn't need it too often.
